pkg/pulumi/pulumi.go

```diff
@@ -170,3 +177,45 @@ func deployCharts(configPath string) pulumi.RunFunc {
 		return nil
 	}
 }
+
+func ensureStateDirExists(statePath string) error {
+	// Specifically handle ~ as a prefix
+	if strings.HasPrefix(statePath, "~") {
```

Comment:
nitpicking here, but this is not correct, we don't want to expand `~foo`, but rather only `~/foo`. In fact `~foo` should be expanded to `foo` user home directory

e.g. https://stackoverflow.com/questions/17609732/expand-tilde-to-home-directory

Reply:
Thanks for the catch!